### PR TITLE
ElmLintBear: Remove `pragma: no cover`

### PR DESCRIPTION
**Pull request by yash-nisar**
Remove `pragma: no cover` in `check_prerequisites` by adding
a suitable test.

Closes https://github.com/coala/coala-bears/issues/1996
Related to https://github.com/coala/coala-bears/issues/1618

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by yash-nisar** [comment]
CC @jayvdb @Makman2

File: tests/elm/ElmLintBearTest.py [hunk 1]
```diff
@@ -143,3 +144,18 @@ def test_good_function(self):
                 [],
                 filename=fname,
                 create_tempfile=False)
+
+    def test_check_prerequisites(self):
+        _shutil_which = shutil.which
+        try:
+            shutil.which = lambda *args, **kwargs: None
```
Comment (Makman2) [inline tests/elm/ElmLintBearTest.py]: use `@patch` on the test method. You can patch the `which` you imported from `shutil`, so the patch-module-string gets `bears.elm.ElmLintBear.which`. Then you don't need to change the import inside `ElmLintBear`, you just have to remove the pragma.

Comment (yash-nisar) [inline tests/elm/ElmLintBearTest.py]: Cool, on it. :+1:

**Comment by yash-nisar** [comment]
Done @Makman2

**Comment by Nosferatul** [comment]
I think this pr is ready, what do you say @Makman2 ?

**Comment by Makman2** [comment]
yes looks good :)

**Comment by Makman2** [comment]
ack 9d2563a

**Comment by yash-nisar** [comment]
reack b1e3fbd9de6ab719edc30ad4ef344c6e6363b553

**Comment by yash-nisar** [comment]
I'm doing a minor change -> Replacing `assertTrue` with `assertEqual` as in the docs.
>  This method should also be avoided when more specific methods are available (e.g. assertEqual(a, b) instead of assertTrue(a == b)), because they provide a better error message in case of failure.

**Comment by yash-nisar** [comment]
reack 0a25352953ee2b3d91e1350d50153e495fc203f2

**Comment by yash-nisar** [comment]
ack 0a25352953ee2b3d91e1350d50153e495fc203f2

**Comment by yukiisbored** [comment]
@rultor please merge

**Comment by yash-nisar** [comment]
Thanks @yukiisbored :smile:

**Comment by Makman2** [comment]
> I'm doing a minor change -> Replacing assertTrue with assertEqual as in the python docs.
>
> This method should also be avoided when more specific methods are available (e.g. assertEqual(a, b) instead of assertTrue(a == b)), because they provide a better error message in case of failure.

Actually the docs want you to avoid such operators like `==` to use in `assertTrue`. If your result should be true in the end, `assertTrue` is just effective^^ however like this is also okay^^

**Comment by Makman2** [comment]
But if you do a change, don't ack it yourself, it has to be reviewed again by someone else.

**Comment by yash-nisar** [comment]
Sure, will keep that in mind. I felt the need to change it because I missed a bug in my travis PR. Also, I've opened [an issue](https://github.com/coala/coala-bears/issues/1998) with regard to this that will avoid future bugs 😉 :

> In general, self.assertTrue('some_string') does not fail. So even if check_prerequisites() returns 'You are not connected to the internet.', the test will pass.
This can be avoided either by explicitly doing self.assertTrue(URLBear.check_prerequisites() == True) or by self.assertEqual(URLBear.check_prerequisites(), True). The latter one is more preferred.

**Comment by Makman2** [comment]
> Actually the docs want you to avoid such operators like == to use in assertTrue. If your result should be true in the end, assertTrue is just effective^^ however like this is also okay^^

Nvm this, `assertEqual(x, True)` and `assertTrue(x)` are two different things. `assertTrue` invokes `bool` on the given expression, while `assertEqual` does not.

**Comment by yash-nisar** [comment]
Yeah, I agree :+1: . Just wanted to make sure things like these -> `self.assertTrue('some_string')` do not pass. :wink:

**Pull request merged**